tests/integration/test_execution_environment.py

```diff
@@ -1,11 +1,7 @@
-"""test the use of execution-environment through to runner
-"""
+"""Test the use of ``execution-environment`` through to runner."""
```

Comment:
Could you move the docstring updates to a separate PR? It's quite hard to focus on other changes with these in the diff.

Reply:
I prefer them in the same because adding the parameters describes what is being passed around and what the modified functions and files do. And doing 3 PRs, one to update the current doc strings, one to modify the tests, and one to update them after would be a bit much IMO

Reply:
BTW, if the scope of the doc string changes extended outside of the files and functions being modified, I would agree here.

Reply:
It'd be 2 PRs — one to update the docstrings, the second one injecting new args into tests and adding them to the docstrings in one line, not changing the entire docstrings.

Reply:
> BTW, if the scope of the doc string changes extended outside of the files and functions being modified, I would agree here.

The PR title says that `pytest-mock` is added but instead there are docstring changes that don't even mention this. It makes the PR non-atomic and the description is inaccurate.

Reply:
updated

Reply:
Please try to look past the scope concern. IMO, the value of the change significantly outweighs the scope question here.

Reply:
I'm trying, it's just that these changes partially shadow the other ones which sometimes end up becoming invisible/unnonicable to the reviewers.